types/react-router/index.d.ts

```diff
@@ -64,7 +64,7 @@ export interface RouteComponentProps<P> {
 
 export interface RouteProps {
   location?: H.Location;
-  component?: React.SFC<RouteComponentProps<any> | undefined> | React.ComponentClass<RouteComponentProps<any> | undefined>;
+  component?: React.SFC<RouteComponentProps<any> & any | undefined> | React.ComponentClass<RouteComponentProps<any> & any | undefined>;
```

Comment:
@minestarks is right. Adding `any` to possible types means removing type constraint at all, therefore this update should not be merged.